### PR TITLE
Fix tx deserialisation for adding nonce

### LINKED ISSUE #704: Issue with solana Versioned Transaction
**Describe the bug**
At present, the solana crypto helper has trouble signing a Versioned transaction.

This is demonstrated with the following transaction;

```python
txn = {'signatures': [[1], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], 'message': [128, {'header': {'numRequiredSignatures': 1, 'numReadonlySignedAccounts': 0, 'numReadonlyUnsignedAccounts': 8}, 'accountKeys': [[15], [139, 243, 6, 84, 129, 61, 49, 97, 16, 104, 201, 21, 75, 85, 128, 199, 51, 117, 98, 236, 173, 28, 166, 79, 160, 148, 183, 105, 125, 236, 55, 33], [10, 5, 36, 65, 161, 88, 78, 87, 140, 112, 5, 246, 167, 190, 30, 5, 116, 25, 119, 87, 112, 15, 36, 25, 0, 242, 95, 74, 253, 138, 25, 206], [80, 45, 184, 150, 44, 155, 78, 165, 230, 208, 65, 158, 223, 160, 22, 229, 254, 56, 67, 165, 115, 5, 16, 29, 238, 192, 204, 187, 10, 112, 1, 28], [117, 144, 35, 85, 67, 187, 212, 221, 93, 0, 171, 53, 106, 46, 173, 245, 66, 126, 74, 140, 90, 210, 66, 90, 148, 40, 246, 48, 1, 181, 20, 73], [215, 231, 89, 241, 109, 176, 189, 157, 215, 63, 157, 185, 99, 214, 43, 185, 87, 46, 112, 113, 226, 234, 43, 101, 72, 215, 18, 65, 192, 118, 103, 73], [251, 128, 139, 123, 204, 4, 145, 168, 183, 14, 87, 24, 42, 206, 54, 83, 150, 37, 161, 115, 200, 241, 128, 157, 59, 108, 121, 167, 217, 230, 164, 74], [251, 173, 2, 185, 182, 133, 15, 4, 115, 222, 105, 67, 253, 28, 21, 33, 16, 55, 114, 133, 103, 18, 175, 115, 175, 130, 78, 84, 120, 106, 253, 236], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [3, 6, 70, 111, 229, 33, 23, 50, 255, 236, 173, 186, 114, 195, 155, 231, 188, 140, 229, 187, 197, 247, 18, 107, 44, 67, 155, 58, 64, 0, 0, 0], [4, 121, 213, 91, 242, 49, 192, 110, 238, 116, 197, 110, 206, 104, 21, 7, 253, 177, 178, 222, 163, 244, 142, 81, 2, 177, 205, 162, 86, 188, 19, 143], [6, 221, 246, 225, 215, 101, 161, 147, 217, 203, 225, 70, 206, 235, 121, 172, 28, 180, 133, 237, 95, 91, 55, 145, 58, 140, 245, 133, 126, 255, 0, 169], [140, 151, 37, 143, 78, 36, 137, 241, 187, 61, 16, 41, 20, 142, 13, 131, 11, 90, 19, 153, 218, 255, 16, 132, 4, 142, 123, 216, 219, 233, 248, 89], [180, 63, 250, 39, 245, 215, 246, 74, 116, 192, 155, 31, 41, 88, 121, 222, 75, 9, 171, 54, 223, 201, 221, 81, 75, 50, 26, 167, 179, 140, 229, 232], [207, 197, 49, 175, 119, 226, 174, 207, 211, 166, 26, 30, 84, 144, 202, 47, 98, 152, 139, 62, 50, 42, 93, 88, 209, 223, 254, 181, 103, 251, 79, 92], [226, 248, 119, 232, 61, 174, 198, 136, 143, 67, 171, 181, 174, 123, 49, 27, 183, 140, 52, 150, 8, 151, 252, 229, 167, 184, 44, 22, 226, 15, 241, 151]], 'recentBlockhash': [152, 168, 73, 110, 90, 226, 78, 23, 213, 59, 213, 211, 225, 116, 15, 94, 203, 198, 236, 203, 177, 49, 205, 35, 94, 118, 161, 135, 143, 163, 59, 34], 'instructions': [[7], {'programIdIndex': 8, 'accounts': [[0]], 'data': [[5], 2, 192, 92, 21, 0]}, {'programIdIndex': 8, 'accounts': [[0]], 'data': [[9], 3, 160, 15, 0, 0, 0, 0, 0, 0]}, {'programIdIndex': 11, 'accounts': [[6], 0, 2, 0, 19, 7, 10], 'data': [[1], 1]}, {'programIdIndex': 7, 'accounts': [[2], 0, 2], 'data': [[12], 2, 0, 0, 0, 56, 91, 1, 0, 0, 0, 0, 0]}, {'programIdIndex': 10, 'accounts': [[1], 2], 'data': [[1], 17]}, {'programIdIndex': 9, 'accounts': [[25], 10, 14, 0, 2, 1, 6, 5, 19, 13, 9, 9, 12, 9, 20, 10, 14, 15, 1, 18, 6, 16, 17, 3, 4, 21], 'data': [[37], 193, 32, 155, 51, 65, 214, 156, 129, 0, 1, 0, 0, 0, 17, 1, 100, 0, 1, 56, 91, 1, 0, 0, 0, 0, 0, 16, 172, 1, 0, 0, 0, 0, 0, 10, 0, 0]}, {'programIdIndex': 10, 'accounts': [[3], 2, 0, 0], 'data': [[1], 9]}], 'addressTableLookups': [[1], {'accountKey': [170, 106, 49, 103, 152, 193, 186, 121, 187, 208, 2, 242, 232, 128, 22, 18, 251, 219, 59, 40, 197, 230, 28, 71, 136, 76, 238, 230, 100, 21, 109, 76], 'writableIndexes': [[4], 105, 108, 107, 102], 'readonlyIndexes': [[3], 159, 104, 103]}]}]}

payer = SolanaCrypto("./solana_private_key.txt")
payer.sign_transaction(txn)
*** KeyError: 'recentBlockhash'
```

The signing doesnt know what to do with the 'recentBlockhash' key which is populated as so in orignal python code;

```
        recent_blockhash = self.ledger_api.api.get_latest_blockhash().value.blockhash
        txn["message"][1]["recentBlockhash"] = json.loads(recent_blockhash.to_json())
        msg = MessageV0.from_json(json.dumps(txn["message"][1]))
        # i have no idea what this is doing
        signed_transaction = VersionedTransaction(msg, [payer.entity])
        # i dont know how this has generated us a signed transaction, but it has.
```

Once i have received permission to share the private repo, I will do so which will give a cleaner idea and some tests to demonstrate working functionality.

**To Reproduce**
2. Do '....'
3. See error '...'

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - AEA Version [e.g. 1.0.0]

**Additional context**

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

fixes #704

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: plugins/aea-ledger-solana/aea_ledger_solana/helper.py [hunk 1]
```diff
@@ -231,19 +231,6 @@ def generate_tx_nonce(seller: Address, client: Address) -> str:
...
         return aggregate_hash.hexdigest()
 
-    def add_nonce(self, tx: dict) -> JSONLike:
-        """
-        Check whether a transaction is valid or not.
-
-        :param tx: the transaction.
-        :return: True if the random_message is equals to tx['input']
-        """
-        stxn = SolanaTransaction.from_json(tx)
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/helper.py:241]: Deserialisation will fail if transaction is versioned

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 2]
```diff
@@ -208,6 +208,19 @@ def create_pda(
...
         return json.loads(tx)
 
+    def add_nonce(self, tx: dict) -> JSONLike:
+        """
+        Check whether a transaction is valid or not.
+
+        :param tx: the transaction.
+        :return: True if the random_message is equals to tx['input']
+        """
+        stxn = self.deserialize_tx(tx=tx)
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/solana.py:218]: This will fix the issue

Comment (Dassy23) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:218]: Lgtm

**Review by Dassy23 — COMMENTED** [review]
Lgtm

**Review by Dassy23 — APPROVED** [review]

**Pull request merged**